### ISSUE TITLE
AMPI_Ireduce only creates a request at the root

### ISSUE THREAD

**Issue by stwhite91** [UIUC-PPL/charm#1376]
*Original issue: https://charm.cs.illinois.edu/redmine/issues/1376*

---
MPI_Ireduce and other "all-to-one" type nonblocking collectives currently only create and set the request on the root rank. The MPI standard says that all ranks should be able to wait on the resulting request, so we just need to create a pre-completed request elsewhere.

**Comment by stwhite91** [comment]
*Original date: 2017-01-23 03:21:14*

---
examples/ampi/creduce only waits on the result at the root. That should be changed as well.

**Comment by PhilMiller** [comment]
*Original date: 2017-02-12 02:09:31*

---
Was the code not actually wrong, or did this get fixed? 'Rejected' without an explanation is rather hard to follow.

**Comment by stwhite91** [comment]
*Original date: 2017-02-13 03:02:18*

---
I don't remember marking this rejected? It is a real issue, so reopening...

**Comment by stwhite91** [comment]
*Original date: 2017-02-13 19:53:56*

---
~~https://charm.cs.illinois.edu/gerrit/#/c/2239/~~ https://github.com/UIUC-PPL/charm/commit/dad73e6fcbc71386959102cf3af16ff00b1a7be5
